esmvalcore/cmor/check.py

```diff
@@ -196,39 +186,45 @@ def check_metadata(self, logger=None):
 
         return self._cube
 
-    def check_data(self, logger=None):
+    def check_data(self, logger: Optional[logging.Logger] = None) -> Cube:
         """Check the cube data.
 
-        Performs all the tests that require to have the data in memory.
         Assumes that metadata is correct, so you must call check_metadata prior
         to this.
 
         It will also report some warnings in case of minor errors.
 
         Parameters
         ----------
-        logger: logging.Logger
+        logger:
             Given logger.
 
+        Returns
+        -------
+        iris.cube.Cube
+            Checked cube.
+
         Raises
         ------
         CMORCheckError
             If errors are found. If fail_on_error attribute is set to True,
             raises as soon as an error is detected. If set to False, it perform
             all checks and then raises.
+
         """
         if logger is not None:
             self._logger = logger
 
-        if self._cmor_var.units:
-            units = self._get_effective_units()
-            if str(self._cube.units) != units:
-                self._cube.convert_units(units)
+        # TODO: remove in v2.12
+        if self.automatic_fixes:
+            self._cube = self._automatic_fix.fix_data(self._cube)
 
         self._check_coords_data()
 
+        self.report_debug_messages()
```

Comment:
Yes, this is a tiny bugfix - `check_data` did not previously report debug messages. There aren't any at the moment, that's why no one noticed, but we should have this call here in case someone adds a debug message at some point.